workaround_buggy_warnings_as_errors_return_value.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/sh
+
+workaround_buggy_warnings_as_errors_return_value() {
+  _TEMP_FILE=$(mktemp mix_compile_stderr.XXXXXXXX)
+
+  # deletes all build artifacts for the current project to ensure they are
+  # actually compiled and applicable warnings generated
+  mix clean
+
+  # prevents warnings in dependencies from being confused with warnings in the current project
+  mix deps.compile
+
+  # the following pair of lines emulate the behaviour of tee for stderr
+  mix compile 2> "$_TEMP_FILE"
+  cat "$_TEMP_FILE" 1>&2
```

Comment:
For colored output:
```suggestion
  script -qc "mix compile" "$_TEMP_FILE"
```
Reference: [linux - How to write stdout to file with colors? - Stack Overflow](https://stackoverflow.com/questions/27397865/how-to-write-stdout-to-file-with-colors)